catalog/video/video-security/src/test/java/org/codice/alliance/video/security/claims/videographer/VideographerClaimsHandlerTest.java

```diff
@@ -160,14 +161,96 @@ public boolean matches(Object o) {
         assertThat(claimsCollection, containsInAnyOrder(nameClaim, emailClaim, ipClaim));
 
         claimsParameters = new ClaimsParameters();
+        claimsParameters.setPrincipal(mock(VideographerPrincipal.class));
         claimsCollection = claimsHandler.retrieveClaimValues(requestClaims, claimsParameters);
 
         assertThat(claimsCollection, hasSize(2));
 
+    }
+
+    @SuppressWarnings("unchecked")
+    @Test
+    public void testRetrieveClaimsForNonVideographerPrincipal() throws URISyntaxException {
+        VideographerClaimsHandler claimsHandler = new VideographerClaimsHandler();
+        claimsHandler.setAttributes(Arrays.asList(CLAIM1,
+                CLAIM_URI_2 + "=" + CLAIM_VALUE_2A + "|" + CLAIM_VALUE_2B + "|" + CLAIM_VALUE_2C,
+                CLAIM3));
+
+        ClaimCollection requestClaims = new ClaimCollection();
+        Claim requestClaim = new Claim();
+        URI nameURI = new URI(CLAIM_URI_1);
+        requestClaim.setClaimType(nameURI);
+        requestClaims.add(requestClaim);
+        requestClaim = new Claim();
+        URI emailURI = new URI(CLAIM_URI_2);
+        requestClaim.setClaimType(emailURI);
+        requestClaims.add(requestClaim);
+        requestClaim = new Claim();
+        URI fooURI = new URI("http://schemas.xmlsoap.org/ws/2005/05/identity/claims/foobar");
+        requestClaim.setClaimType(fooURI);
+        requestClaim.setOptional(true);
+        requestClaims.add(requestClaim);
+
+        ClaimsParameters claimsParameters = new ClaimsParameters();
+        claimsParameters.setPrincipal(new VideographerPrincipal("127.0.0.1"));
+
+        List<URI> supportedClaims = claimsHandler.getSupportedClaimTypes();
+
+        assertThat(supportedClaims, hasSize(3));
```

Comment:
Should we move this to its own separate test? It would help keep the tests shorter and make it easier to figure out what's broken when a test fails.